pandas/_libs/internals.pyx

```diff
@@ -890,12 +890,12 @@ cdef class BlockValuesRefs:
 
     def __cinit__(self, blk: Block | None = None) -> None:
         if blk is not None:
-            self.referenced_blocks = [weakref.ref(blk)]
+            self.referenced_blocks = [PyWeakref_NewRef(blk, None)]
```

Comment:
Does this actually change anything? I am under the impression that Cython would be smart enough to handle this the same in both cases.

Instead of looking at the benchmarks (which can be difficult to run and flaky), you can also inspect the Cython annotations before/after the change. Are they actually different?

Reply:
Thanks. Glad you brought that up. I should've been more explicit in the description, as the motivation for this PR is that the code-gen changes quite dramatically. An example from one of the methods below. Note the code generated for the method does not fit on one screen before the change
## Before
<img width="1410" alt="image" src="https://github.com/user-attachments/assets/9eb22c7d-309a-40a2-b637-cd7e1f282648">

## After
<img width="1675" alt="image" src="https://github.com/user-attachments/assets/23738417-3d5b-4d2b-bc82-a25d82cf7640">

Reply:
Thanks for posting. That's...quite the difference